runtime/src/main/java/com/dylibso/chicory/runtime/Module.java

```diff
@@ -106,11 +107,37 @@ public Instance instantiate() {
         if (hostImports == null) {
             hostImports = new HostImports();
         }
-        return this.instantiate(hostImports, initialize, start, listener);
+        return this.instantiate(hostImports, InterpreterMachine::new, initialize, start, listener);
+    }
+
+    public Instance instantiate(Function<Instance, Machine> machineFactory) {
+        return this.instantiate(new HostImports(), machineFactory, true, true, null);
+    }
+
+    public Instance instantiate(HostImports hostImports) {
+        return this.instantiate(hostImports, true, true);
+    }
+
+    public Instance instantiate(
+            HostImports hostImports, Function<Instance, Machine> machineFactory) {
+        return this.instantiate(hostImports, machineFactory, true, true, null);
+    }
+
+    public Instance instantiate(
```

Comment:
I recently cleaned up a little this class to make it mandatory to use the Builder Pattern.
This should be `withMachineFactory` and remove the other `instantiate` overloads.